omegaconf/base.py

```diff
@@ -391,53 +413,28 @@ def resolve_simple_interpolation(
                 else:
                     return None
 
-    def _resolve_interpolation(
+    def resolve_interpolation(
```

Comment:
1. keep this _private, I don't care about the opinion of some generic tool telling me that I shouldn't call private functions in my library from my library (the warnings should go away after a rebase, but wait with the rebase until I am done with the first review pass.

2. It looks like there isn't a real need to split _resolve_interpolation into two functions. If this is correct please combine them.

Reply:
1. Done in 40976e8

2. Do you mean the `_resolve_complex` vs `_resolve_simple` split? That split is actually needed I think: the first one resolves using the grammar, while the second one is "lower level" and is used to resolve the node / resolver interpolations (assuming access to their resolved names / arguments)

Reply:
1. great
2. Okay, hard for me to address this after 19 days without re-reviewing. letting this one go for now, it might resurface on a final review pass.

Reply:
> 2. Okay, hard for me to address this after 19 days without re-reviewing. letting this one go for now, it might resurface on a final review pass.

Sure, just to give the high-level reasoning: there is one resolve function that we call to resolve any expression that may contain interpolations (this is `_resolve_complex()`). This function parses the expression, and in particular identifies where the interpolations are. Then to obtain the final result we need to be able to actually resolve the value of these interpolations, and this is when we call `_resolve_simple()`: this function doesn't do any parsing, its role is to either fetch the value of a key (for an interpolation of the form `${foo.bar}`) or to call a resolver (for an interpolation of the form `${foo:bar}`)